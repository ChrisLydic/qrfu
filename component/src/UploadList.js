import React from 'react';
import PropTypes from 'prop-types';

// example components for displaying uploaded files

export default function UploadList(props) {
    const items = props.uploads.map(file => {
        if (file.type === 'audio') {
            return (<li key={file.name}>
                        <audio src={`${file.url}`} controls></audio>
                        <div>{file.name}</div>

                        {props.delete &&
                            <button className="qr-delete" type="button" onClick={() => props.delete(file.name)}><img src="/delete.png" alt="delete file"/></button>
                        }
                    </li>);
        } else {
            return (<li key={file.name}>
                        <a href={`${file.url}`}>
                            <img src={`${file.url}`} alt='preview' />
                            <div>{file.name}</div>
                        </a>

                        {props.delete && 
                            <button className="qr-delete" type="button" onClick={() => props.delete(file.name)}><img src="/delete.png" alt="delete file"/></button>
                        }
                    </li>);
        }
    });
    
    if (props.uploads.length > 1 && props.delete) {
        return (
            <div>
                <button className="qr-delete-all" type="button" onClick={() => {props.uploads.forEach(f => props.delete(f.name))}}>
                    Delete All
                </button>

                <ul className='custom'>{items}</ul>
            </div>
        );
    } else {
        return (
            <ul className='custom'>{items}</ul>
        );
    }
}

export function LinkList(props) {
    const items = props.uploads.map(file => 
        (<li key={file.name}>
            <a href={`${file.url}`}>
                {file.name}
            </a>
            {props.delete &&
                <button className="qr-delete" type="button" onClick={() => props.delete(file.name)}><img src="/delete.png" alt="delete file"/></button>
            }
        </li>)
    );
    
    if (props.uploads.length > 1 && props.delete) {
        return (
            <div>
                <button className="qr-delete-all" type="button" onClick={() => {props.uploads.forEach(f => props.delete(f.name))}}>
                    Delete All
                </button>

                <ul>{items}</ul>
            </div>
        );
    } else {
        return (
            <ul>{items}</ul>
        );
    }
}

export function ProfileImage(props) {
    if (props.uploads.length > 0) {
        let file = props.uploads[props.uploads.length - 1];

        return (
            <ul className='custom'>
                <li key={file.name}>
                    <a href={`${file.url}`}>
                        <img src={`${file.url}`} alt='preview' />
                        <div>{file.name}</div>
                    </a>

                    {props.delete && 
                        <button className="qr-delete" type="button" onClick={() => props.delete(file.name)}><img src="/delete.png" alt="delete file"/></button>
                    }
                </li>
            </ul>
        );
    } else {
        return (
            <ul className='custom'></ul>
        );
    }
}

UploadList.propTypes = LinkList.propTypes = ProfileImage.propTypes = {
    uploads: PropTypes.arrayOf(
        PropTypes.shape({
            name: PropTypes.string.isRequired,
            url: PropTypes.string.isRequired,
            uid: PropTypes.string,
            type: PropTypes.string,
        })
    ).isRequired,
    delete: PropTypes.func,
};