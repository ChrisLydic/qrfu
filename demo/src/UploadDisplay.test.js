import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import UploadDisplay from './UploadDisplay';
import UploadList from './UploadList';

test('UploadDisplay loads with QR code', () => {
    const url = '127.0.0.1:3000/ui/123';
    const uid = '123';

    const component = renderer.create(
        <UploadDisplay
				uid={uid}
				url={url}
				uploadlist={UploadList}
				uploads={[]}
                isError={false} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('UploadDisplay loads without QR code', () => {
    const url = '';
    const uid = '';

    const component = renderer.create(
        <UploadDisplay
				uid={uid}
				url={url}
				uploadlist={UploadList}
				uploads={[]} 
                isError={false} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('UploadDisplay loads with error', () => {
    const url = '';
    const uid = '';

    const component = renderer.create(
        <UploadDisplay
				uid={uid}
				url={url}
				uploadlist={UploadList}
				uploads={[]} 
                isError={true} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});