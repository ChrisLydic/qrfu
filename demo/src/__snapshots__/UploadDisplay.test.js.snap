// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadDisplay loads with QR code 1`] = `
<div
  className="code"
>
  <canvas
    height={128}
    style={
      Object {
        "height": 128,
        "width": 128,
      }
    }
    width={128}
  />
  <ul
    className="custom"
  />
</div>
`;

exports[`UploadDisplay loads without QR code 1`] = `
<div
  className="code"
>
  <ul
    className="custom"
  />
</div>
`;
